src/main.ts

```diff
@@ -301,6 +302,42 @@ export class Purchases {
     return toOfferings(offeringsResponse, productsResponse);
   }
 
+  private async getSingleOffering(
+    offeringsResponse: OfferingsResponse,
+    appUserId: string,
+    offeringIdentifier: string,
+    params?: GetOfferingsParams,
+  ): Promise<Offering | null> {
+    const isCurrentOffering =
+      offeringsResponse.current_offering_id === offeringIdentifier;
+    const offeringResponse = offeringsResponse.offerings.find(
```

Comment:
Nit for readability:
```suggestion
    const selectedOfferingResponse = offeringsResponse.offerings.find(
```